### PR TITLE
Simplify Idempotence Capability

### PR DESCRIPTION
**Pull request by the-real-adammork**
Soundtrack of this PR: [Idris Muhammed - Could Heaven Ever Be Like This](https://www.youtube.com/watch?v=0skigTyA4NE)

### Motivation

The requirements for Transaction Idempotence have changed, and now only require a Seed to be passed down into the Transaction Builder.

### In this PR
* Add RngSeed type so consumers can pass in a seed to the Transaction Builder
* Remove RNG seedability for txo selection
* Update unit tests
* Add new comparison utils for testing if TxOuts are idempotent (same txOutPublicKey)

### CODE REVIEW

File: Tests/Integration/Network/NetworkConfigFixtures.swift
Comment (plarson) [inline Tests/Integration/Network/NetworkConfigFixtures.swift:13]: Does this affect CI?

**Review by plarson — APPROVED** [review]

**Review by stevenspiel — APPROVED** [review]

File: Tests/Integration/Network/NetworkConfigFixtures.swift
Comment (the-real-adammork) [reply Tests/Integration/Network/NetworkConfigFixtures.swift:13]: no effect on CI, and should be testNet as thats the most stable test network

File: Sources/MobileCoinClient+Deprecated.swift
Comment (alanpaulin) [inline Sources/MobileCoinClient+Deprecated.swift:231]: Is this for backwards compatibility so that Moby can integrate the updated SDK without simultaneously updating to use the new API? 👏

File: Sources/Transaction/Outputs/PendingTransaction.swift
Comment (alanpaulin) [inline Sources/Transaction/Outputs/PendingTransaction.swift:44]: Nice

File: Sources/Transaction/Outputs/PendingTransaction.swift
Comment (carybakker) [inline Sources/Transaction/Outputs/PendingTransaction.swift]: given the meaning of idempotent, it's not 100% clear, by name alone, if this should compare for equality or inequality...   

Can we rename these to `isIdempotentlyEqual` or something like that? 

Same for the `are` functions...

**Review by alanpaulin — APPROVED** [review]

**Review by carybakker — APPROVED** [review]
LGTM!

File: Sources/Transaction/Outputs/PendingTransaction.swift
Comment (alanpaulin) [reply Sources/Transaction/Outputs/PendingTransaction.swift]: I like the name `isIdempotentlyEqual` or `isIdempotentlyEquivalent`

Comment (the-real-adammork) [reply Sources/Transaction/Outputs/PendingTransaction.swift]: I was thinking to add "with" so it would read `isIdempotent(with: lhs)` call-site:

```swift
payload.isIdempotent(with: otherPayload)
```

I think the equal qualifier is redundant with idempotent. Not sure the static functions gain much from a rename.

File: Sources/MobileCoinClient+Deprecated.swift
Comment (the-real-adammork) [reply Sources/MobileCoinClient+Deprecated.swift:231]: ya, it was a public API for a small amount of time so should be deprecated before removal.

File: Sources/Transaction/Outputs/PendingTransaction.swift
Comment (the-real-adammork) [reply Sources/Transaction/Outputs/PendingTransaction.swift]: changed to `isIdempotent(with: lhs)`

**Comment by the-real-adammork** [comment]
will merge when CI passes

File: Sources/Transaction/Outputs/PendingTransaction.swift
Comment (the-real-adammork) [reply Sources/Transaction/Outputs/PendingTransaction.swift]: `isSomething` implies test for equality to "something", `isNotSomething` is the reverse.

**Comment by the-real-adammork** [comment]
putting the merge on hold, until I finish discussing with Bernie.

**Comment by the-real-adammork** [comment]
good for now.

**Comment by the-real-adammork** [comment]
The requirements for Transaction Idempotence have changed, and now only require a Seed to be passed down into the Transaction Builder.

* Add RngSeed type so consumers can pass in a seed to the Transaction Builder
* Remove RNG seedability for txo selection
* Update unit tests
* Add new comparison utils for testing if TxOuts are idempotent (same txOutPublicKey)

**Pull request merged**
